src/utils/crypto.ts

```diff
@@ -6,10 +6,14 @@
  */
 
 import CryptoJS from 'crypto-js';
+import bitcore from 'bitcore-lib';
 import { DecryptionError, InvalidPasswdError, UnsupportedHasherError } from '../errors';
-import { HASH_ITERATIONS, HASH_KEY_SIZE } from '../constants';
+import { HATHOR_MAGIC_BYTES, HASH_ITERATIONS, HASH_KEY_SIZE } from '../constants';
 import { IEncryptedData } from '../types';
 
+// Monkey-patch MAGIC_BYTES to use Hathor's
+bitcore.Message.MAGIC_BYTES = Buffer.from(HATHOR_MAGIC_BYTES);
+
```

Comment:
Do we know why bitcore uses these magic bytes?
And does bitcore do anything else to the data other than prefix with the magic bytes?
I say this because we may have to implement this signature and verification on other places that cannot use bitcore's library (e.g. Ledger app).

So we may have to understand the whole signature process and be able to duplicate it on other places using other languages.

Reply:
From my investigation, it is a convention that was [first proposed by bitcoin-core](https://github.com/bitcoin/bitcoin/pull/524) and that bitcore-lib also uses

The idea of having the message prefixed with an known string is to prevent someone from tricking a user into signing a transaction instead of a simple text message

Looking at the [implementation](https://github.com/bitpay/bitcore-message/blob/master/lib/message.js), I don't see anything preventing us from implementing it on Ledger